src/App.tsx

```diff
@@ -36,10 +19,26 @@ function App() {
             path: '/socket.io',
           })
         : io('http://localhost:3001');
-    socket.on('playerConnect', (id) => addClient(id));
-    socket.on('playerDisconnect', (id) => removeClient(id));
+    socket.on('playerConnect', (clientIds) => {
+      setClients(clientIds);
+    });
+    socket.on('playerDisconnect', (clientIds) => {
+      setClients(clientIds);
+    });
+    socket.on('playerAssignment', (playerId: string) => {
```

Comment:
see in the server code this `playerAssignment` was emitted from the server with `socket.emit()`. That will emit it only to the sender, so not everyone gets assigned that id, just the connected client that it belongs to